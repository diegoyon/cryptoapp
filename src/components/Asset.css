.asset {
  display: flex;
  position: relative;
  height: 8rem;
  flex-direction: column;
  justify-content: flex-end;
  padding: 0.5rem;
}

.asset h1 {
  position: absolute;
  left: 1rem;
  top: 1rem;
  align-self: center;
  color: #2d4573;
  font-size: 3.5rem;
}

.basic-detail {
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  color: white;
  z-index: 2;
}

.basic-detail h2 {
  font-size: 1.3rem;
}
