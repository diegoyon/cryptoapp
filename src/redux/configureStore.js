import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import assetsReducer from './Assets';

const store = configureStore({
  reducer: {
    assets: assetsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
