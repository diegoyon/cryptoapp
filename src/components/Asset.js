import React from 'react';
import PropTypes from 'prop-types';
import './Asset.css';

function Asset(props) {
  const { symbol, name, priceUsd } = props;
  let price;
  if (priceUsd > 10) {
    price = parseFloat(priceUsd).toLocaleString(undefined, {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    });
  } else {
    price = Number(parseFloat(priceUsd).toFixed(5));
  }
  return (
    <div className="asset">
      <h1 data-testid="symbol">{symbol}</h1>
      <div className="basic-detail">
        <h2 data-testid="name">{name.toUpperCase()}</h2>
        <p>
          $
          {price}
        </p>
      </div>
    </div>
  );
}

Asset.propTypes = {
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  priceUsd: PropTypes.string.isRequired,
};

export default Asset;
