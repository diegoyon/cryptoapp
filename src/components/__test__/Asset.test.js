import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import renderer from 'react-test-renderer';
import Asset from '../Asset';

describe(Asset, () => {
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Asset
          symbol="BTC"
          name="Bitcoin"
          priceUsd="20000"
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('asset displays correct name', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Asset
          symbol="BTC"
          name="Bitcoin"
          priceUsd="20000"
        />
      </Provider>,
    );
    const assetName = getByTestId('name').textContent;
    expect(assetName).toBe('BITCOIN');
  });

  it('asset displays correct symbol', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Asset
          symbol="BTC"
          name="Bitcoin"
          priceUsd="20000"
        />
      </Provider>,
    );
    const assetSymbol = getByTestId('symbol').textContent;
    expect(assetSymbol).toBe('BTC');
  });
});
