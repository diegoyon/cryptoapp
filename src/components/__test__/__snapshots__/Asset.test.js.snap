// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset renders correctly 1`] = `
<div
  className="asset"
>
  <h1
    data-testid="symbol"
  >
    BTC
  </h1>
  <div
    className="basic-detail"
  >
    <h2
      data-testid="name"
    >
      BITCOIN
    </h2>
    <p>
      $
      20,000.00
    </p>
  </div>
</div>
`;
